### PR TITLE
docs: add dnsaddr protocol spec

### PR DESCRIPTION
**Pull request by alanshaw**
...to the best of my knowledge.

Added as separate doc in `protocols` dir so existing link in README becomes not 404 :)

### CODE REVIEW

File: protocols/DNSADDR.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# `dnsaddr`
+
+`dnsaddr` is a protocol that instructs the resolver to lookup multiaddr(s) in DNS TXT records for the domain name in it's value section.
+
+To resolve a `dnsaddr` multiaddr, the domain name in the value section must first be prefixed with `_dnsaddr.`. Then a DNS query to lookup TXT records for the domain must be made. There may be multiple DNS TXT records for the domain. Valid `dnsaddr` TXT records begin with `dnsaddr=`, followed by a single multiaddr. Recursive lookups are allowed.
+
+Example:
+
+`/dnsaddr/bootstrap.libp2p.io`
```
Comment (jacobheun) [inline protocols/DNSADDR.md]:
```suggestion
`/dnsaddr/bootstrap.libp2p.io` would result in a query of `_dnsaddr.bootstrap.libp2p.io`.
```

File: protocols/DNSADDR.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# `dnsaddr`
+
+`dnsaddr` is a protocol that instructs the resolver to lookup multiaddr(s) in DNS TXT records for the domain name in it's value section.
+
+To resolve a `dnsaddr` multiaddr, the domain name in the value section must first be prefixed with `_dnsaddr.`. Then a DNS query to lookup TXT records for the domain must be made. There may be multiple DNS TXT records for the domain. Valid `dnsaddr` TXT records begin with `dnsaddr=`, followed by a single multiaddr. Recursive lookups are allowed.
+
+Example:
```
Comment (lidel) [inline protocols/DNSADDR.md]:
```suggestion
## Example
```

**Comment by olizilla** [comment]
fun note: we had to add the 2 levels of indirection because we couldn't fit 8 records (ip4 & ip6 per boostrap node) into a single udp packet for the dns response https://github.com/protocol/bifrost-infra/issues/338

**Comment by lidel** [comment]
@olizilla mind adding a PR against `protocols/DNSADDR.md` with UDP limitation you discovered? (in "Caveats" section or something) :pray:  I am pretty sure it will save someone a ton of time in low-level debug :eyes:

File: protocols/DNSADDR.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# `dnsaddr`
```
Comment (Stebalien) [inline protocols/DNSADDR.md]: @alanshaw 

This should probably explain suffix matching: Given the address `/dnsaddr/bootstrap.libp2p.io/SOMETHING`, this address will only resolve to multiaddrs that end with SOMETHING.

**Pull request merged**